code/integration/main/main.ino

```diff
@@ -2,52 +2,91 @@
-//Important: DON'T include SoftwareSerial.h it is already included by WiFiService.h
-
 #include <Arduino.h>
 #include "WiFiService.h"
 #include "driver_timer.h"
-// Create WiFiService object
-WiFiService myWiFiService;
+#include "common.h"
+#include "Parser.h"
+
+/******************************************************************************************************************/
+// include files
+WiFiService myWiFiService;      //!< Create WiFiService object
+Parser myParser;                //!< Create Parser object
+STATES g_states;                //!< states for state machine
 
+/****************************************************************************************************************//*
+   \brief     initialization
+/******************************************************************************************************************/
 void setup()
 {
-  
-  Serial.begin(9600);     // Set data rate for the HW serial port
-  while (!Serial) {;}     // wait for HW serial port to connect.
+  Serial.begin(9600);           // set data rate for the HW serial port
+  while (!Serial){;}            // init HW serial port wait for HW serial port to connect.
 
-  if( setupTimer() == true )
+  if ( setupTimer() == true )   // init timer module
   {
     Serial.print("TimerSetupSuccessful\r\n" );
   }
   else Serial.print("TimerSetupFailed\r\n" );
-}
-
 
+  myWiFiService.Init();         // init Wifi class
+}
 
-void loop() {
 
-  myWiFiService.Init(false);
-  timerRuntime();                     // example output timer
-  
-  if (myWiFiService.String_Is_Complete())
+/****************************************************************************************************************//*
+   \brief     main loop
+/******************************************************************************************************************/
+void loop()
+{
+  myWiFiService.Run(true);                                              // continous wifi check
+  delay( 1000 );
+  if ( myWiFiService.String_Is_Complete() )                             // package received
   {
-    Serial.println("");
-    Serial.print("String detected: ");
-    Serial.println(myWiFiService.Get_String());
-    myWiFiService.ResetString();
-    
-    //DEBUGGING CODE
-    if (debug_WiFiService) {Serial.println("");Serial.println("MainLoop - if String_Is_Complete");delay(1000);}
-  }
+    timerRuntime();                                                     // example output timer
 
+    if (myWiFiService.String_Is_Complete())
+    {
+      Serial.println("");
+      Serial.print("String detected: ");
+      
+      String received_string = myWiFiService.Read();                    // get string from Wifly
+      g_states = myParser.RunParser(received_string, 0, 0);             // interpret string
+      String parser_return_string = myParser.Get_String_from_Parser();  // get string for factory
+      
+      switch ( g_states )                                               // evaluate next steps
```

Comment:
@izabellasandor ja timer runtime wird nur aufgerufen wenn ein string detektiert wird. Du hast recht, es sollte in die erste Zeile. Aendere ich.